File: api/src/HamptonHawksPlantSales.Core/Models/Customer.cs
namespace HamptonHawksPlantSales.Core.Models;

public class Customer : BaseEntity
{
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string DisplayName { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string PickupCode { get; set; } = string.Empty;
    public string? Notes { get; set; }
}

File: web/src/types/customer.ts
export interface Customer {
  id: string;
  firstName: string | null;
  lastName: string | null;
  displayName: string;
  phone: string | null;
  email: string | null;
  pickupCode: string;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCustomerRequest {
  firstName?: string | null;
  lastName?: string | null;
  displayName: string;
  phone?: string | null;
  email?: string | null;
  notes?: string | null;
}
